runtime/interpreter/interpreter.go

```diff
@@ -355,6 +355,8 @@ type Interpreter struct {
 	atreeValueValidationEnabled    bool
 	atreeStorageValidationEnabled  bool
 	tracingEnabled                 bool
+	inStorageIteration             bool
```

Comment:
Move both  `inStorageIteration` and `storageMutatedDuringIteration` to `sharedState`